lib/methods.js

```diff
@@ -115,5 +117,17 @@ module.exports = {
     return this.reverse().filter(function(elem, index, arr) {
       return arr.indexOf(elem, index + 1) === -1;
     }).reverse();
+  },
+
+  zip: function(/*...args*/) {
+    var args = Array.prototype.slice.call(arguments);
+    var allArrays = [this].concat(args).filter(utilities.isArray);
+    var allLengths = allArrays.map(function (array) { return array.length; });
+    var highestLength = Math.max.apply(null, allLengths);
+    var result = [];
+    for(var i = 0; i < highestLength; i++) {
```

Comment:
i didn't think it made sense to use `.forEach()` because it's not an iteration of any particular array in the group, just a stepwise operation up to the length of the longest array. along the same lines, we're not directly interested in the elements of a particular array, just the index at that step.